employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java

```diff
@@ -99,87 +108,143 @@ private void refresh() {
+                                penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore().toShortString() + ").");
                     } else {
                         indictmentIcons.add(IconContainer.Icon.ROTATION_VIOLATION_PENALTY, "Shift is unassigned although it has a Rotation Employee (" +
-                                                                                           penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore() + ").");
+                                penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore().toShortString() + ").");
                     }
                 }
-                
+
                 for (UnassignedShiftPenalty penalty : shiftView.getUnassignedShiftPenaltyList()) {
-                    indictmentIcons.add(IconContainer.Icon.UNASSIGNED_SHIFT_PENALTY, "Shift is unassigned (Score: " + penalty.getScore() + ").");
+                    indictmentIcons.add(IconContainer.Icon.UNASSIGNED_SHIFT_PENALTY, "Shift is unassigned (Score: " + penalty.getScore().toShortString() + ").");
                 }
             }
         }
     }
 
+    private String getNiceString(Shift shift) {
+        return shift.getSpot().getName() + ":" + getNiceString(shift.getStartDateTime()) + " to " + getNiceString(shift.getEndDateTime());
+    }
+
+    private String getNiceString(OffsetDateTime dateTime) {
+        DateTimeFormat timeFormat = DateTimeFormat.getFormat(PredefinedFormat.DATE_TIME_SHORT);
+        return timeFormat.format(GwtJavaTimeWorkaroundUtil.toDate(GwtJavaTimeWorkaroundUtil.toLocalDateTime(dateTime)));
+    }
+
+    private void updateColor(HardMediumSoftLongScore score) {
+        if (score.getHardScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getHardScore(), 5, 0, 10));
+        } else if (score.getHardScore() > 0) {
+            setAnimationDelay(mapToRange(score.getHardScore(), 85, 100, 10));
+        } else if (score.getMediumScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getMediumScore(), 30, 15, 1));
+        } else if (score.getMediumScore() > 0) {
+            setAnimationDelay(mapToRange(score.getMediumScore(), 70, 85, 10));
+        } else if (score.getSoftScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getSoftScore(), 50, 30, 100));
+        } else if (score.getSoftScore() > 0) {
+            setAnimationDelay(mapToRange(score.getSoftScore(), 50, 70, 30));
+        } else {
+            setAnimationDelay(50);
+        }
+    }
+
+    // NOTE: if amount == 0, then it returns start
+    //       if amount >= maxAmount, then it returns end
+    //       else it returns start + amount * ((end - start)/maxAmount)
+    private double mapToRange(double amount, double start, double end, double maxAmount) {
+        if (amount < 0) {
+            throw new IllegalArgumentException("amount (" + amount + ") must be non-negative.");
```

Comment:
Iddeally, a sentence (as it ends with a ".") should start with a capital, so I 'd say "The amount ("...

Reply:
not blocking the merge

Reply:
Will address this in a future PR